packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -503,6 +498,13 @@ export function createRouteManifest(
 		...[...fileBasedRoutes, ...injectedRoutes, ...redirectRoutes].sort(routeComparator),
 	];
 
+	settings.buildOutput = getPrerenderDefault(config) ? 'static' : 'server';
+
+	// Check if
+	for (const route of routes) {
+		await getRoutePrerenderOption(route, settings, params.fsMod, logger);
```

Comment:
I think we should probably do this in parallel to avoid blocking reading each route one at a time.